wayland-display-core/src/wayland/handlers/compositor.rs

```diff
@@ -129,7 +128,7 @@ impl CompositorHandler for State {
         }
 
         if let Some(popup) = self.popups.find_popup(surface) {
-            let PopupKind::Xdg(ref popup) = popup;
+            let PopupKind::Xdg(ref popup) = popup else { todo!("refutable pattern in local binding") };
```

Comment:
Ah right, we also now have input method handling in the popup code. Our compositor doesn't do that, so you can just replace the `todo!` with `unreachable!()`, so it will (most likely) get optimized away.